PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetPerformance.cxx

```diff
@@ -1886,53 +1886,56 @@ void AliAnalysisTaskEmcalJetPerformance::ComputeBackground()
       }
     }
     
-    // Loop over clusters. Sum the cluster ET:
+    // Loop over clusters, if the jet container is for full jets. Sum the cluster ET:
     // (1) in the EMCal, (2) in the EMCal fiducial volume, (3) in the DCal, (4), in the DCal fiducial volume, (5) in the EMCal random cone, (6) in the DCal random cone
-    AliClusterContainer* clusCont = GetClusterContainer(0);
-    AliTLorentzVector clus;
-    Double_t clusEta;
-    Double_t clusPhi;
-    Double_t clusPt;
-    for (auto clusIterator : clusCont->accepted_momentum() ) {
-      
-      clus.Clear();
-      clus = clusIterator.first;
-      clusEta = clus.Eta();
-      clusPhi = clus.Phi_0_2pi();
-      clusPt = clus.Pt();
-      
-      // (1)
-      if (TMath::Abs(clusEta) < etaEMCal && clusPhi > phiMinEMCal && clusPhi < phiMaxEMCal) {
-        clusPtSumEMCal += clusPt;
-      }
-      
-      // (2)
-      if (TMath::Abs(clusEta) < etaEMCalfid && clusPhi > phiMinEMCalfid && clusPhi < phiMaxEMCalfid) {
-        clusPtSumEMCalfid += clusPt;
-      }
-      
-      // (3)
-      if (TMath::Abs(clusEta) > etaMinDCal && TMath::Abs(clusEta) < etaEMCal && clusPhi > phiMinDCal && clusPhi < phiMaxDCal) {
-        clusPtSumDCal += clusPt;
-      }
-      
-      // (4)
-      if (TMath::Abs(clusEta) > etaMinDCalfid && TMath::Abs(clusEta) < etaEMCalfid && clusPhi > phiMinDCalfid && clusPhi < phiMaxDCalfid) {
-        clusPtSumDCalfid += clusPt;
-      }
-      
-      // (5)
-      deltaR = GetDeltaR(&clus, etaEMCalRC, phiEMCalRC);
-      if (deltaR < jetR) {
-        clusPtSumEMCalRC += clusPt;
-      }
-      
-      // (6)
-      deltaR = GetDeltaR(&clus, etaDCalRC, phiDCalRC);
-      if (deltaR < jetR) {
-        clusPtSumDCalRC += clusPt;
+    if (jetCont->GetArrayName().Contains("Full")) {
```

Comment:
It may not be super efficient to have TString::Contains() in your event loop, but I don't think it's a huge deal (if you want to change it, maybe you could check if the cluster container attach to the jet is null?)

Reply:
True, was just out of laziness. Your suggestion is anyway more robust, I'll make a fix. Thanks